src/neo/Ledger/Blockchain.cs

```diff
@@ -398,25 +415,14 @@ private void OnNewHeaders(Header[] headers)
             system.TaskManager.Tell(new TaskManager.HeaderTaskCompleted(), Sender);
         }
 
-        private void OnNewTransaction(Transaction transaction, bool relay)
+        private VerifyResult OnNewTransaction(Transaction transaction)
         {
-            RelayResultReason reason;
-            if (ContainsTransaction(transaction.Hash))
-                reason = RelayResultReason.AlreadyExists;
-            else if (!MemPool.CanTransactionFitInPool(transaction))
-                reason = RelayResultReason.OutOfMemory;
-            else
-                reason = transaction.Verify(currentSnapshot, MemPool.SendersFeeMonitor.GetSenderFee(transaction.Sender));
-
-            if (reason == RelayResultReason.Succeed)
-            {
-                if (!MemPool.TryAdd(transaction.Hash, transaction))
-                    reason = RelayResultReason.OutOfMemory;
-                else if (relay)
-                    system.LocalNode.Tell(new LocalNode.RelayDirectly { Inventory = transaction });
-            }
-
-            Sender.Tell(reason);
+            if (ContainsTransaction(transaction.Hash)) return VerifyResult.AlreadyExists;
+            if (!MemPool.CanTransactionFitInPool(transaction)) return VerifyResult.OutOfMemory;
```

Comment:
It would be great if we could manipulate this order of verification, @erikzhang.

In moments which the mempool has plenty of pending txs we verify this and line 424 first, otherwise, they are the last to be verified.

Reply:
L424 is an atomic operation and you cannot separate it.